Source/WebCore/dom/Element.cpp

```diff
@@ -3639,6 +3639,9 @@ ExceptionOr<RefPtr<Attr>> Element::setAttributeNode(Attr& attrNode)
         if (attributeNodeValue.hasException())
             return attributeNodeValue.releaseException();
         attrNodeValue = AtomString(attributeNodeValue.releaseReturnValue());
+
+        if (!attributeTypeAndSink.attributeType.isNull() && attrNode.ownerElement() && attrNode.ownerElement() != this)
```

Comment:
Why do you need to check that ownerElement is non-null? Presumably this comparison can handle that?

Reply:
Ah that's a good point. I'll drop that check.

Reply:
Ah wait we do want that check. We only want to throw if it has an owner that's not this. If it has no owner that's fine.